### PR TITLE
Adrien/feature/added twitter to reddit

### PR DESCRIPTION
**Pull request by why-not-try-calmer**
Continuation of #29. 

To do:
- add tests
- add documentation

### CODE REVIEW

File: defrag/__init__.py [hunk 1]
```diff
@@ -33,5 +34,17 @@
 NO_LOAD = []
 LOAD = []
 
+
+def pretty_log(comment: str, value: Optional[str] = None, warn: bool = False) -> None:
```
Comment (KaratekHD) [inline defrag/__init__.py:38]: Why do we need this? Isn't `LOGGER` enough?

File: defrag/modules/twitter.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                              access_token_key=env["TWITTER_ACCESS_TOKEN"], access_token_secret=env["TWITTER_ACCESS_TOKEN_SECRET"])
+
+            fetch = as_async(api.GetUserTimeline)
+            entries: List[Any] = await fetch(screen_name="@openSUSE")
+            return sort_make(entries)
+        except Exception as err:
+            print("Unable to fetch from Twitter @openSUSE: ", err)
+
+
+def register_service():
```
Comment (KaratekHD) [inline defrag/modules/twitter.py:53]: I am a bit confused, where do you call this?

File: defrag/__init__.py (hunk 1 above)
Comment (why-not-try-calmer) [reply defrag/__init__.py:38]: Have you tried to see the difference?

File: defrag/modules/twitter.py (hunk 2 above)
Comment (why-not-try-calmer) [reply defrag/modules/twitter.py:53]: Wherever we want! Check out the unittests for an example, the new documentation under this function from my latest commit for more details, and the docs of the ServiceManager class for the bigger picture.

**Comment by why-not-try-calmer** [comment]
Okay, I think this PR is now worth considering for a merge. But before that, maybe you have a clue why the unittests for reddit and twitter fail (404) even though I am getting the correct result when querying the app locally from a web browser?

**Review by KaratekHD — APPROVED** [review]

**Comment by KaratekHD** [comment]
> Okay, I think this PR is now worth considering for a merge. But before that, maybe you have a clue why the unittests for reddit and twitter fail (404) even though I am getting the correct result when querying the app locally from a web browser?

I can not test that at the moment, I need to get API keys for the services first. Not sure if I can get that done before our meeting though, since I also need to prepare our vacation.

**Comment by why-not-try-calmer** [comment]
> > Okay, I think this PR is now worth considering for a merge. But before that, maybe you have a clue why the unittests for reddit and twitter fail (404) even though I am getting the correct result when querying the app locally from a web browser?
> 
> I can not test that at the moment, I need to get API keys for the services first. Not sure if I can get that done before our meeting though, since I also need to prepare our vacation.

Okay, thanks for the approval, no problem for testing, I'll just leave it as it. We'll merge later.

**Pull request merged**